drivers/ph_oem/ph_oem.c

```diff
@@ -0,0 +1,446 @@
+    i2c_read_reg(I2C, ADDR, PH_OEM_REG_UNLOCK, &reg_value, 0x0);
+
+    if (reg_value != 0x00) {
+        DEBUG("\n[ph_oem debug] Failed at unlocking I2C address register. \n");
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_i2c_address(ph_oem_t *dev, uint8_t addr)
+{
+    if (ph_oem_unlock_address_reg(dev) != PH_OEM_OK) {
+        return PH_OEM_WRITE_ERR;
+    }
+
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_ADDRESS, addr, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting I2C address to %x failed\n", addr);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+
+    dev->params.addr = addr;
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_enable_interrupt(ph_oem_t *dev, ph_oem_interrupt_pin_cb_t cb,
+                            void *arg, ph_oem_irq_option_t option)
+{
+    if (dev->params.interrupt_pin == GPIO_UNDEF) {
+        return PH_OEM_INTERRUPT_GPIO_UNDEF;
+    }
+
+    int gpio_flank = 0;
+    int gpio_mode = 0;
+
+    dev->arg = arg;
+    dev->cb = cb;
+
+    /* gpio_mode might be different for your microcontroller. Just GPIO_IN
+     * didn't worked with Zolertia Firefly, because pin was so sensitive the
+     * IRQ was constantly called */
+    switch (option) {
+        case PH_OEM_IRQ_DISABLED:
+            break;
+        case PH_OEM_IRQ_FALLING:
+            gpio_flank = GPIO_FALLING;
+            gpio_mode = GPIO_IN_PU;
+            break;
+        case PH_OEM_IRQ_RISING:
+            gpio_flank = GPIO_RISING;
+            gpio_mode = GPIO_IN_PD;
+            break;
+        case PH_OEM_IRQ_BOTH:
+            gpio_flank = GPIO_BOTH;
+            gpio_mode = GPIO_IN_PD;
+            break;
+    }
+
+    if (option != PH_OEM_IRQ_DISABLED) {
+        if (gpio_init_int(dev->params.interrupt_pin,
+                          gpio_mode, gpio_flank, cb, arg) < 0) {
+
+            DEBUG("\n[ph_oem debug] Initilizing enable gpio pin failed.\n");
+            return PH_OEM_GPIO_INIT_ERR;
+        }
+    }
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_interrupt_pin(const ph_oem_t *dev, ph_oem_irq_option_t option)
+{
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_INTERRUPT, option, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting interrupt pin to option %d failed.\n", option);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_led_state(const ph_oem_t *dev, ph_oem_led_state_t state)
+{
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_LED, state, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting LED state to %d failed.\n", state);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_enable_device(const ph_oem_t *dev, bool enable)
```

Comment:
I don't really think the enable pin should be included in the device driver as it is not a functionality provided by the actual device and needs external setup. This should be handled in the application.

Reply:
Yes you are right, it makes not sense. I removed all lines related to the enable pin